Build failure: cdemu

### Steps To Reproduce

Steps to reproduce the behavior:
1. build cdemu/vhba on latest nixos-unstable

### Build log

```
vhba> /build/vhba-module-20240202/vhba.c:1087:15: error: initialization of 'void (*)(struct platform_device *)' from incompatible pointer type 'int (*)(struct platform_device *)' [-Werror=incompatible-pointer-types]
vhba>  1087 |     .remove = vhba_remove,
vhba>       |               ^~~~~~~~~~~
vhba> /build/vhba-module-20240202/vhba.c:1087:15: note: (near initialization for 'vhba_platform_driver.<anonymous>.remove')
kwin-better-blur> -- Found KF6Config: /nix/store/bkxfqm4zigarz5k4rc58mzmjx964nbbw-kconfig-6.6.0-dev/lib/cmake/KF6Config/KF6ConfigConfig.cmake (found version "6.6.0")
kwin-better-blur> -- Found KF6ConfigWidgets: /nix/store/yin77vn55ngw1dpprm56crzz3mjm6idh-kconfigwidgets-6.6.0-dev/lib/cmake/KF6ConfigWidgets/KF6ConfigWidgetsConfig.cmake (found version "6.6.0")
vhba> cc1: all warnings being treated as errors
kwin-better-blur> -- Found KF6CoreAddons: /nix/store/1kf9xmlm2bgy2jllbms9y3w8vr6b85la-kcoreaddons-6.6.0-dev/lib/cmake/KF6CoreAddons/KF6CoreAddonsConfig.cmake (found version "6.6.0")
vhba> make[3]: *** [/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/source/scripts/Makefile.build:244: /build/vhba-module-20240202/vhba.o] Error 1
vhba> make[2]: *** [/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/source/Makefile:1926: /build/vhba-module-20240202] Error 2
vhba> make[1]: *** [/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/source/Makefile:224: __sub-make] Error 2
vhba> make[1]: Leaving directory '/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/build'
vhba> make: *** [Makefile:14: modules] Error 2
error: builder for '/nix/store/9r6hwr3mkcs80hl8gx23d1xpj0y27z42-vhba-20240202.drv' failed with exit code 2;
       last 23 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/xrx8sw2fri58sw11pdqr7k260x77va7d-vhba-module-20240202.tar.xz
       > source root is vhba-module-20240202
       > setting SOURCE_DATE_EPOCH to timestamp 1706870626 of file vhba-module-20240202/vhba.c
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       > build flags: SHELL=/nix/store/izpf49b74i15pcr9708s3xdwyqs4jxwl-bash-5.2p32/bin/bash O=\$\(buildRoot\) CC=/nix/store/zznja5f8v3jafffyah1rk46vpfcn38dv-gcc-wrapper-13.3.0/bin/cc HOSTCC=/nix/store/zznja5f8v3jafffyah1rk46vpfcn38dv-gcc-wrapper-13.3.0/bin/cc HOSTLD=/nix/store/b74nxf0yn2dzha02mgdxyklaqjaijzqp-binutils-wrapper-2.42/bin/ld ARCH=x86_64 KDIR=/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/build INSTALL_MOD_PATH=\$\(out\)
       > make -C /nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/build M=/build/vhba-module-20240202 modules
       > make[1]: Entering directory '/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/build'
       >   CC [M]  /build/vhba-module-20240202/vhba.o
       > /build/vhba-module-20240202/vhba.c:1087:15: error: initialization of 'void (*)(struct platform_device *)' from incompatible pointer type 'int (*)(struct platform_device *)' [-Werror=incompatible-pointer-types]
       >  1087 |     .remove = vhba_remove,
       >       |               ^~~~~~~~~~~
       > /build/vhba-module-20240202/vhba.c:1087:15: note: (near initialization for 'vhba_platform_driver.<anonymous>.remove')
       > cc1: all warnings being treated as errors
       > make[3]: *** [/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/source/scripts/Makefile.build:244: /build/vhba-module-20240202/vhba.o] Error 1
       > make[2]: *** [/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/source/Makefile:1926: /build/vhba-module-20240202] Error 2
       > make[1]: *** [/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/source/Makefile:224: __sub-make] Error 2
       > make[1]: Leaving directory '/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/build'
       > make: *** [Makefile:14: modules] Error 2
       For full logs, run 'nix log /nix/store/9r6hwr3mkcs80hl8gx23d1xpj0y27z42-vhba-20240202.drv'.
error: 1 dependencies of derivation '/nix/store/5d881pnr6vrz5p14p2dhi7c1vjacyfi9-linux-6.11-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/d4w29j0c23kbpkcl4z2ki0grq2idw6cl-nixos-system-nixos-desktop-24.11.20240916.99dc878.drv' failed to build
┏━ 1 Errors:
 ⋮
┃        > unpacking source archive /nix/store/xrx8sw2fri58sw11pdqr7k260x77va7d-vhba-module-20240202.tar.xz
┃        > source root is vhba-module-20240202
┃        > setting SOURCE_DATE_EPOCH to timestamp 1706870626 of file vhba-module-20240202/vhba.c
┃        > Running phase: patchPhase
┃        > Running phase: updateAutotoolsGnuConfigScriptsPhase
┃        > Running phase: configurePhase
┃        > no configure script, doing nothing
┃        > Running phase: buildPhase
┃        > build flags: SHELL=/nix/store/izpf49b74i15pcr9708s3xdwyqs4jxwl-bash-5.2p32/bin/bash O=\$\(buildRoot\) CC=/nix/store/zznja5f8v3jafffyah1rk46vpfcn38dv-gcc-wrapper-13.3.0/bin/cc HOSTCC=/nix/store/zznja5f8v3jafffyah1rk46vp…
┃        > make -C /nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/build M=/build/vhba-module-20240202 modules
┃        > make[1]: Entering directory '/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/build'
┃        >   CC [M]  /build/vhba-module-20240202/vhba.o
┃        > /build/vhba-module-20240202/vhba.c:1087:15: error: initialization of 'void (*)(struct platform_device *)' from incompatible pointer type 'int (*)(struct platform_device *)' [-Werror=incompatible-pointer-types]
┃        >  1087 |     .remove = vhba_remove,
┃        >       |               ^~~~~~~~~~~
┃        > /build/vhba-module-20240202/vhba.c:1087:15: note: (near initialization for 'vhba_platform_driver.<anonymous>.remove')
┃        > cc1: all warnings being treated as errors
┃        > make[3]: *** [/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/source/scripts/Makefile.build:244: /build/vhba-module-20240202/vhba.o] Error 1
┃        > make[2]: *** [/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/source/Makefile:1926: /build/vhba-module-20240202] Error 2
┃        > make[1]: *** [/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/source/Makefile:224: __sub-make] Error 2
┃        > make[1]: Leaving directory '/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/build'
┃        > make: *** [Makefile:14: modules] Error 2
┃        For full logs, run 'nix log /nix/store/9r6hwr3mkcs80hl8gx23d1xpj0y27z42-vhba-20240202.drv'.
```

### Additional context

Looks like there's a new vhba module version: https://sourceforge.net/projects/cdemu/files/vhba-module/

### Notify maintainers
@bendlas 
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.10.10, NixOS, 24.11 (Vicuna), 24.11.20240913.345c263`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/l3amk5lsakpc93him5kry24kax23sn4h-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/emulators/cdemu/daemon.nix
+++ b/pkgs/applications/emulators/cdemu/daemon.nix
@@ -2,9 +2,9 @@
 stdenv.mkDerivation {
 
   inherit (callPackage ./common-drv-attrs.nix {
-    version = "3.2.6";
+    version = "3.2.7";
     pname = "cdemu-daemon";
-    hash = "sha256-puQE4+91xhRuNjVPZYgN/WO0uO8fVAOdxQWOGQ+FfY8=";
+    hash = "sha256-EKh2G6RA9Yq46BpTAqN2s6TpLJb8gwDuEpGiwdGcelc=";
   }) pname version src meta;
 
   nativeBuildInputs = [ cmake pkg-config intltool ];

--- a/pkgs/applications/emulators/cdemu/libmirage.nix
+++ b/pkgs/applications/emulators/cdemu/libmirage.nix
@@ -5,9 +5,9 @@
 stdenv.mkDerivation {
 
   inherit (callPackage ./common-drv-attrs.nix {
-    version = "3.2.7";
+    version = "3.2.9";
     pname = "libmirage";
-    hash = "sha256-+okkgNeVS8yoKSrQDy4It7PiPlTSiOsUoFxQ1FS9s9M=";
+    hash = "sha256-JBd+wHSZRyRW1SZsaAaRO2dNUFkpwRCr3s1f39KyWIs=";
   }) pname version src meta;
 
   PKG_CONFIG_GOBJECT_INTROSPECTION_1_0_GIRDIR = "${placeholder "out"}/share/gir-1.0";

--- a/pkgs/applications/emulators/cdemu/vhba.nix
+++ b/pkgs/applications/emulators/cdemu/vhba.nix
@@ -2,11 +2,11 @@
 
 stdenv.mkDerivation rec {
   pname = "vhba";
-  version = "20240202";
+  version = "20240917";
 
   src  = fetchurl {
     url = "mirror://sourceforge/cdemu/vhba-module-${version}.tar.xz";
-    hash = "sha256-v1hQ1Lj1AiHKh9c0OpKe2oexkfb1roxhQXRUO1ut3oM=";
+    hash = "sha256-zjTLriw2zvjX0Jxfa9QtaHG5tTC7cLTKEA+WSCP+Dpg=";
   };
 
   makeFlags = kernel.makeFlags ++ [ "KDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build" "INSTALL_MOD_PATH=$(out)" ];

